apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { StyleSheet, Text, Icon } from '@kiwicom/universal-components';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
+
+import type { VisaDetail as VisaDetailType } from './__generated__/VisaDetail.graphql';
+
+type Props = {|
+  +data: ?VisaDetailType,
+|};
+
+type VisaRequiredType = {
+  +visaRequired: ?boolean,
+};
+
+const isVisaRequired = (
+  passengers: $ReadOnlyArray<?VisaRequiredType>,
+): boolean => passengers.some(passenger => passenger?.visaRequired);
+
+const VisaDetail = (props: Props) => {
+  const passengers = props.data?.passengers;
+  const visaRequired = passengers && isVisaRequired(passengers);
+  const color = visaRequired
+    ? defaultTokens.colorIconCritical
+    : defaultTokens.colorIconSuccess;
+  const textType = visaRequired ? 'critical' : 'success';
+  const text = visaRequired ? 'Visa is required' : 'Visa is not required';
+
+  return (
+    <View
+      style={[
+        styles.wrapperVisa,
+        visaRequired ? styles.critical : styles.success,
```

Comment:
Since you extracted most of the logic before the `return` statement, you could do the same with this line.